src/adapters/conversant.js

```diff
@@ -0,0 +1,230 @@
+  var w = window,
+    n = navigator;
+
+  // production endpoint
+  var conversantUrl = '//media.msg.dotomi.com/s2s/header?callback=$$PREBID_GLOBAL$$.conversantResponse';
+
+  // SSAPI returns JSONP with window.pbjs.conversantResponse as the cb
+  var appendScript = function (code) {
+    var script = document.createElement('script');
+    script.type = 'text/javascript';
+    script.className = 'cnvr-response';
+
+    try {
+      script.appendChild(document.createTextNode(code));
+      document.getElementsByTagName('head')[0].appendChild(script);
+    } catch (e) {
+      script.text = code;
+      document.getElementsByTagName('head')[0].appendChild(script);
+    }
+  };
+
+  var httpPOSTAsync = function (url, data) {
+    var xmlHttp = new w.XMLHttpRequest();
+
+    xmlHttp.onload = function () {
+      appendScript(xmlHttp.responseText);
+    };
+    xmlHttp.open('POST', url, true); // true for asynchronous
+    xmlHttp.withCredentials = true;
+    xmlHttp.send(data);
+  };
+
+  var getDNT = function () {
+    return n.doNotTrack === '1' || w.doNotTrack === '1' || n.msDoNotTrack === '1' || n.doNotTrack === 'yes';
+  };
+
+  var getDevice = function () {
+    return {
+      h: screen.height,
+      w: screen.width,
+      dnt: getDNT() ? 1 : 0,
+      language: n.language.split('-')[0],
+      make: n.vendor ? n.vendor : '',
+      ua: n.userAgent
+    };
+  };
+
+  var callBids = function (params) {
+    var conversantBids = params.bids || [];
+    requestBids(conversantBids);
+  };
+
+  var requestBids = function (bidReqs) {
+    // build bid request object
+    var page = location.pathname + location.search + location.hash,
+      siteId = '',
+      conversantImps = [],
+      conversantBidReqs,
+      secure = 0;
+
+    //build impression array for conversant
+    utils._each(bidReqs, function (bid) {
+      var bidfloor = utils.getBidIdParamater('bidloor', bid.params),
+        sizeArrayLength = bid.sizes.length,
+        adW = 0,
+        adH = 0,
+        imp;
+
+      secure = utils.getBidIdParamater('secure', bid.params) ? 1 : secure;
+      siteId = utils.getBidIdParamater('site_id', bid.params);
+
+      if (sizeArrayLength === 2 && typeof bid.sizes[0] === 'number' && typeof bid.sizes[1] === 'number') {
+        adW = bid.sizes[0];
+        adH = bid.sizes[1];
+      } else {
+        adW = bid.sizes[0][0];
+        adH = bid.sizes[0][1];
+      }
+
+      imp = {
+        id: bid.bidId,
+        banner: {
+          w: adW,
+          h: adH
+        },
+        secure: secure,
+        bidfloor: bidfloor ? bidfloor : 0,
+        displaymanager: 'Prebid.js',
+        displaymanagerver: VERSION
+      };
+
+      conversantImps.push(imp);
+    });
+
+    conversantBidReqs = {
+      'id': utils.getUniqueIdentifierStr(),
+      'imp': conversantImps,
+
+      'site': {
+        'id': siteId,
+        'mobile': document.querySelector('meta[name="viewport"][content*="width=device-width"]') !== null ? 1 : 0,
```

Comment:
The mobile field is there to determine if the site's layout can be optimized when on mobile clients. Not necessarily that they are mobile